colour/io/tm2714.py

```diff
@@ -601,6 +601,34 @@ def comments(self, value: Optional[str]):
 
         self._comments = value
 
+    def __eq__(self, other):
+        if isinstance(other, Header_IESTM2714):
+            comps = [
+                self._manufacturer == other.manufacturer,
+                self._catalog_number == other.catalog_number,
+                self._description == other.description,
+                self._document_creator == other.document_creator,
+                self._unique_identifier == other.unique_identifier,
+                self._measurement_equipment == other.measurement_equipment,
+                self._laboratory == other.laboratory,
+                self._report_number == other.report_number,
+                self._report_date == other.report_date,
+                self._document_creation_date == other.document_creation_date,
+                self._comments == other.comments
+            ]
+            return all(comps)
+        return False
+
+    def __ne__(self, other):
+        return not self == other
```

Comment:
I had to double-check the priority of `==` compared to `not`, for clarity, I tend to parentheses:

```python
    def __ne__(self, other):
        return not (self == other)
```